bond/bond.go

```diff
@@ -186,49 +186,48 @@ func setLinksinNetNs(bondConf *bondingConfig, nspath string, releaseLinks bool)
 		linkNames = append(linkNames, s)
 	}
 
-	if netNs, err = ns.GetNS(nspath); err != nil {
+	if podNs, err = ns.GetNS(nspath); err != nil {
 		return fmt.Errorf("failed to open netns %q: %v", nspath, err)
 	}
+	defer podNs.Close()
 
-	if curnetNs, err = ns.GetCurrentNS(); err != nil {
+	if hostNS, err = ns.GetCurrentNS(); err != nil {
 		return fmt.Errorf("failed to get init netns: %v", err)
 	}
 
 	if releaseLinks {
-		if err := netNs.Set(); err != nil {
-			return fmt.Errorf("failed to enter netns %q: %v", netNs, err)
-		}
+		err = moveLinksBetweenNs(linkNames, podNs, hostNS, "host")
+	} else {
+		err = moveLinksBetweenNs(linkNames, hostNS, podNs, "container")
 	}
+	return err
+}
 
-	if len(linkNames) >= 2 { // currently supporting two or more links to one bond
-		for _, linkName := range linkNames {
-			// get interface link in the network namespace
-			link, err := netlink.LinkByName(linkName)
-			if err != nil {
-				return fmt.Errorf("failed to lookup link interface %q: %v", linkName, err)
-			}
-
-			// set link interface down
-			if err = netlink.LinkSetDown(link); err != nil {
-				return fmt.Errorf("failed to down link interface %q: %v", linkName, err)
-			}
+func moveLinksBetweenNs(links []string, from ns.NetNS, to ns.NetNS, toNsName string) error {
+	return from.Do(func(ns.NetNS) error {
+		if len(links) >= 2 { // currently supporting two or more links to one bond
```

Comment:
To improve readability, can you please check the opposite to return a error or continue with the for otherwise?

something like

if len(links) < 2 {
return error
}

for .....

...
return nil

Reply:
changed it :)